lib/FileModel.php

```diff
@@ -43,51 +39,54 @@ public function analyze($currentDirectory) {
 			if ($item instanceof Folder) {
 				continue;
 			}
-			$this->pushToBiggestFilesIfNecessary($item);
+			$this->pushToBiggestFiles($item);
 			$this->analyzeMimeType($item);
+			usort($this->biggestFiles, function($a, $b) {
+			    return $a[1] < $b[1];
+            });
 		}
-		$this->convertBiggestFilesNodeToArray();
+
 	}
 	/**
+     * @var array $biggestFilesSorted
 	 * @return array
 	 */
 	public function getAnalysisReport() {
-		return [
-			'biggest_files' => $this->biggestFiles,
+        $this->analyze($this->userFolder);
+        $biggestFilesSorted = [];
+        if (sizeof($this->biggestFiles)<10) {

```

Comment:
biggest files sayisini parametre olarak al fonksiyona. Default deger olarak 10 ver, Boylece parametresiz yollanirsa 10 olarak baslamis olur.

Reply:
Bunu fonksiyona disaridan parametre olarak degil ama iceride local variable olarak tanimlayip yazacagim, sikinti olursa gozunden kacmasin diye not dusuyorum.

Reply:
Bir de eger ileride show ... items gibi tıklandığında ya da istenen sayı belirtildiğinde o kadar getirecek bir hale dönüştüreceksek bana kalırsa hepsini js tarafına verip gerekli düzenlemeyi orada yapmak daha mantıklı, şu anda sırası değil ama unutmayayım diye bunu da not düşüyorum

Reply:
js tarafi yine bu fonksiyonu cagirmak zorunda, tum dosyalari isteyip istenilen kadarini front-end de suzmek gereksiz data boyutunu arttirir, o fonksiyon parametre olarak almali bunu